taskwarrior_syncall/tw_caldav_utils.py

```diff
@@ -0,0 +1,177 @@
+import traceback
+from datetime import timedelta
+from typing import Dict, List, Optional, Tuple
+from uuid import UUID
+
+from bubop import logger
+from item_synchronizer.types import Item
+
+aliases_tw_caldav_status = {
+    "completed": "completed",
+    "pending": "needs-action",
+    "waiting": "needs-action",
+    "deleted": "cancelled",
+}
+
+aliases_tw_caldav_priority = {
+    "l": 9,
+    "m": 5,
+    "h": 1,
+}
+
+
+def convert_tw_to_caldav(tw_item: Item) -> Item:
+    assert all(
+        i in tw_item.keys() for i in ("description", "status", "uuid")
+    ), "Missing keys in tw_item"
+
+    caldav_item: Item = {}
+
+    caldav_item["summary"] = tw_item["description"]
+    # description
+    caldav_item["description"] = "IMPORTED FROM TASKWARRIOR\n"
+    if "annotations" in tw_item.keys():
+        for i, annotation in enumerate(tw_item["annotations"]):
+            caldav_item["description"] += f"\n* Annotation {i + 1}: {annotation}"
+
+    caldav_item["description"] += "\n"
+    for k in ["uuid"]:
+        caldav_item["description"] += f"\n* {k}: {tw_item[k]}"
+
+    # Status
+    caldav_item["status"] = aliases_tw_caldav_status[tw_item["status"]]
+
+    # Priority
+    if "priority" in tw_item.keys():
+        caldav_item["priority"] = aliases_tw_caldav_priority[tw_item["priority"].lower()]
+
+    # Timestamps
+    if "modified" in tw_item.keys():
+        caldav_item["last-modified"] = tw_item["modified"]
+
+    # Start/due dates
+    # - If given due date -> (start=due-1, end=due)
+    if "due" in tw_item.keys():
+        caldav_item["start"] = tw_item["due"] - timedelta(hours=1)
+        caldav_item["due"] = tw_item["due"]
+
+    if "tags" in tw_item.keys():
+        caldav_item["categories"] = tw_item["tags"]
+
+    return caldav_item
+
+
+def convert_caldav_to_tw(caldav_item: Item) -> Item:
```

Comment:
Absolutely not critical and if too much of a pain ignore this comment  but, in my latest merge, I introduced a `ConcreteItem` class - and I'm using it instead of `Item` which basically is a dict. Maybe you could use that instead of instantiating and passing around an `Item` ?

https://github.com/bergercookie/syncall/blob/master/syncall/concrete_item.py

And I'm using it as the building block (instead of `Item`) for `FilesystemFile` and `FilesystemSide`

https://github.com/bergercookie/syncall/blob/master/syncall/filesystem/filesystem_file.py

https://github.com/bergercookie/syncall/blob/master/syncall/filesystem/filesystem_side.py